Client/src/util/diyDatasets.ts

```diff
@@ -0,0 +1,15 @@
+export function isDiyWdkRecordId(wdkDatasetId: string) {
+  return wdkDatasetId.startsWith('EDAUD_');
+}
+
+export function wdkRecordIdToDiyUserDatasetId(wdkDatasetId: string) {
+  if (!isDiyWdkRecordId(wdkDatasetId)) {
+    throw new Error(`Tried to obtain the user dataset id of an incompatible WDK record with id ${wdkDatasetId}`);
+  }
+
+  return Number(wdkDatasetId.replace(/^EDAUD_/, ''));
+}
+
+export function diyUserDatasetIdToWdkRecordId(userDatasetId: number) {
+  return `EDAUD_${userDatasetId}`;
+}
```

Comment:
Technical debt: these utilities are temporary workarounds for converting between a DIY study's WDK dataset id (which is currently of the form `EDAUD_{userDatasetId}`) and its UD id.

A more robust approach would be to offer a service/model-backed mapping from WDK dataset ids to UD ids - for example, by added a `userDatasetId` attribute to the WDK records for DIY datasets.